backend/kernelCI_app/views/treeView.py

```diff
@@ -14,23 +14,33 @@ def get(self, request):
 
         checkouts = Checkouts.objects.raw(
             """
-            WITH
-                selection AS (
-                    SELECT
-                        start_time
-                    FROM (
-                        SELECT DISTINCT ON (git_commit_hash)
+            WITH selection AS (
+                SELECT
+                    git_repository_url,
+                    git_repository_branch,
+                    MAX(start_time) AS start_time
+                FROM
+                    (
+                        SELECT
+                            git_repository_url,
+                            git_repository_branch,
+                            origin,
                             start_time
                         FROM
                             checkouts
-                        WHERE origin = %s AND start_time  >= TO_TIMESTAMP(%s)
-                        ) AS selection
-                    ORDER BY
-                        start_time DESC
-                )
+                        ORDER BY
+                            start_time DESC
+                    ) AS selection_sorted
+                WHERE
+                    origin = %s AND start_time  >= TO_TIMESTAMP(%s)
```

Comment:
I think we could move the 
WHERE
    origin = %s AND start_time  >= TO_TIMESTAMP(%s)
to inside the selection_sorted query